diff --git a/dist/AnthropicLanguageModel.js b/dist/AnthropicLanguageModel.js
--- a/dist/AnthropicLanguageModel.js
+++ b/dist/AnthropicLanguageModel.js
@@ -901,7 +901,7 @@
             const caller = part.caller;
             const callerInfo = Predicate.isNotNullish(caller) ? {
               type: caller.type,
-              toolId: "tool_id" in caller ? caller.tool_id : undefined
+              toolId: "tool_id" in caller ? caller.tool_id : null
             } : undefined;
             const params = yield* transformToolCallParams(options.tools, part.name, part.input);
             toolCalls.push({
diff --git a/src/AnthropicLanguageModel.ts b/src/AnthropicLanguageModel.ts
--- a/src/AnthropicLanguageModel.ts
+++ b/src/AnthropicLanguageModel.ts
@@ -1356,7 +1356,7 @@
             const callerInfo = Predicate.isNotNullish(caller)
               ? {
                 type: caller.type,
-                toolId: "tool_id" in caller ? caller.tool_id : undefined
+                toolId: "tool_id" in caller ? caller.tool_id : null
               }
               : undefined

